### ISSUE TITLE
Can't add many2many field

### ISSUE THREAD

**Issue by dslpp056193** [directus/app#1420]
# Bug Report

Can't add many2many field

## Steps to Reproduce

1. Open collection and fields
2. Open collection
3. Try to add field "Many2Many"

## Other Context & Screenshots

![image](https://user-images.githubusercontent.com/11080873/54878366-96695300-4e3c-11e9-88aa-9e8a2766bc9c.png)

## Technical Details

- Device: [eg: Desktop, iPhone6, etc]
- OS: [ Windows 10.1803]
- Browser: [ Chrome 68.0.3440.106]
- Directus App: [ 7.0.0]
- Directus API: [ 2.0.0]
- Install Method: [ cloned `master` branch of `directus/directus`]

**Comment by rijkvanzanten** [comment]
I was able to successfully create a many to many on a fresh installation. If possible, could you send over a database dump so I can reproduce your exact setup and check if there's something corrupted in the database that's causing havoc?

**Comment by srjrol** [comment]
I have encountered this issue before.  My solution was to remove all entries in the ```directus_relations``` table between the collections.  These don't get removed when the interface is deleted so if you're trying to re-create it this is likely the issue.

**Comment by am-webdev** [comment]
I'm struggling with the same issue. I've first create a m2m relation and deleted it manually afterwards. Now, it no longer possible to create a new m2m relation even for other existing collection, the same collections or even new collections. 

Here's my "directus_relations" table

```
mysql> select * from directus_relations;
+----+------------------------------+---------------+----------------------+-----------+----------------+
| id | collection_many              | field_many    | collection_one       | field_one | junction_field |
+----+------------------------------+---------------+----------------------+-----------+----------------+
|  1 | directus_activity            | action_by     | directus_users       | NULL      | NULL           |
|  2 | directus_activity_seen       | user          | directus_users       | NULL      | NULL           |
|  3 | directus_activity_seen       | activity      | directus_activity    | NULL      | NULL           |
|  4 | directus_collections_presets | user          | directus_users       | NULL      | NULL           |
|  5 | directus_collections_presets | group         | directus_groups      | NULL      | NULL           |
|  6 | directus_files               | uploaded_by   | directus_users       | NULL      | NULL           |
|  7 | directus_files               | folder        | directus_folders     | NULL      | NULL           |
|  8 | directus_folders             | parent_folder | directus_folders     | NULL      | NULL           |
|  9 | directus_permissions         | group         | directus_groups      | NULL      | NULL           |
| 10 | directus_revisions           | activity      | directus_activity    | NULL      | NULL           |
| 11 | directus_user_roles          | user          | directus_users       | roles     | role           |
| 12 | directus_user_roles          | role          | directus_roles       | users     | user           |
| 13 | directus_users               | avatar        | directus_files       | NULL      | NULL           |
| 14 | directus_fields              | collection    | directus_collections | fields    | NULL           |
| 15 | pages                        | project_id    | projects             | NULL      | NULL           |
| 16 | directus_activity            | id            | pages                | NULL      | NULL           |
| 17 | elements                     | page_id       | pages                | NULL      | NULL           |
| 18 | elements                     | project_id    | projects             | NULL      | NULL           |
| 19 | tests_elements               | tests_id      | tests                | NULL      | elements_id    |
| 20 | tests_elements               | elements_id   | elements             | tests     | tests_id       |
| 21 | elements                     | id            | tests                | NULL      | NULL           |
| 22 | tests_elements               | elements_id   | elements             | tests     | tests_id       |
| 23 | tests_elements               | tests_id      | tests                | NULL      | elements_id    |
+----+------------------------------+---------------+----------------------+-----------+----------------+
23 rows in set (0.00 sec)
```


Everything with "tests_elements" should not be existing anymore because at the moment, the "tests" is not related anyhow to the elements table. 

Maybe this helps you out.

Best regards
Andreas

**Comment by benhaynes** [comment]
@am-webdev — if you deleted it manually, did you also remove it from `directus_fields`?

**Comment by am-webdev** [comment]
hi @benhaynes 
sorry for the misunderstanding. I've used the Directus app to remove the relation by removing the junction collection and the column of the collections "tests" afterwards. I did not ever touched the internal tables. 

So I ran into this issue by simple using the UI provided by default.
If you need more SQL dumps, there is nothing special in my DB so I can provide you the full dump if this might help to isolate the root cause.

**Comment by rijkvanzanten** [comment]
@am-webdev Try removing the old records from directus_relations to see if that helps. I'm closing this as the OP can't be reproduced.

**Comment by am-webdev** [comment]
Hi @rijkvanzanten unfortunately, cleaning the relations table does not fix the issue for me.

So I assume I have to reset the mysql data complete and start from scratch. In my case this is easier as try to fix the issue. I was only waiting if my personal data may helps you to isolate the issue/root cause.

Please let me know if a can assist anyhow. 

Best regards
Andreas

**Comment by rijkvanzanten** [comment]
Could you verify that you're having the same issues on a fresh setup? If so, could you send over a database dump of the fresh setup that's broken? That's the easiest way for me to reproduce this 🙂

**Comment by am-webdev** [comment]
I'll do so and keep you up to date. In addition I'll document the exact steps to reproduce (if it happened again) with a stock fresh installation.

**Comment by am-webdev** [comment]
what I did so far:

1. Fresh database
2. docker-compose up -d (mysql:5.7, directus/app:latest, directus/api:latest)
3. Create new Collection "elements"
4. Create new Collection "tests"
5. Add field "test_elements" to Collection "Tests"
6. Select the Elements Collection within the relation dialog and click "auto-generate"

**Result**: Junction Collection was created successfully

**Now**

1. Open the settings of Collection "Tests"
2. Delete the newly created field "Test Elements"

**Result**: Nothing breaks 💃 

I've removed the junction table manually afterwards and was able to create new many-to-many fields. 
So I'm currently not able to reproduce the behavior. If it appears once more, I send you the sql dump. 

I don't know what special things I did the last time, but now everything is working.
